automation/issuer/bot/issuer_bot.py

```diff
@@ -98,11 +98,12 @@ def handle_handle_registered_investor(event):
             if total_observers == 0:
                 return
 
-            logging.info(f"adding {total_observers} observer(s)")
-            client.submit_exercise(event.cid, 'Token_AddObservers',
-                                   {'party': client.party,
-                                    'newObservers': {'textMap': {**investors, **exchanges, **custodians}}
-                                   })
+            if event.cdata['isPublic']:
```

Comment:
would it make more sense to check this in the beginning of the callback so that you avoid doing all the unnecessary lookups?